sycl/include/sycl/ext/intel/experimental/grf_size_properties.hpp

```diff
@@ -0,0 +1,91 @@
+//==- grf_size_properties.hpp - GRF size kernel properties for Intel GPUs -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===--------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/ext/oneapi/properties/property.hpp>
+#include <sycl/ext/oneapi/properties/property_value.hpp>
+
+#define SYCL_EXT_INTEL_GRF_SIZE 1
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext::intel::experimental {
+struct grf_size_key {
+  template <unsigned int Size>
+  using value_t = oneapi::experimental::property_value<
+      grf_size_key, std::integral_constant<unsigned int, Size>>;
+};
+
+struct grf_size_automatic_key {
+  using value_t = oneapi::experimental::property_value<grf_size_automatic_key>;
+};
+
+template <unsigned int Size>
+inline constexpr grf_size_key::value_t<Size> grf_size;
+
+inline constexpr grf_size_automatic_key::value_t grf_size_automatic;
+
+} // namespace ext::intel::experimental
+namespace ext::oneapi::experimental {
+template <>
+struct is_property_key<sycl::ext::intel::experimental::grf_size_key>
+    : std::true_type {};
+
+template <>
+struct is_property_key<sycl::ext::intel::experimental::grf_size_automatic_key>
+    : std::true_type {};
+
+namespace detail {
+template <>
+struct PropertyToKind<sycl::ext::intel::experimental::grf_size_key> {
+  static constexpr PropKind Kind = PropKind::GRFSize;
+};
+
+template <>
+struct IsCompileTimeProperty<sycl::ext::intel::experimental::grf_size_key>
+    : std::true_type {};
+
+template <>
+struct PropertyToKind<sycl::ext::intel::experimental::grf_size_automatic_key> {
+  static constexpr PropKind Kind = PropKind::GRFSizeAutomatic;
+};
+
+template <>
+struct IsCompileTimeProperty<
+    sycl::ext::intel::experimental::grf_size_automatic_key> : std::true_type {};
+
+template <unsigned int Size>
+struct PropertyMetaInfo<
+    sycl::ext::intel::experimental::grf_size_key::value_t<Size>> {
+  static constexpr const char *name = "sycl-grf-size";
+  static constexpr unsigned int value = Size;
+};
+template <>
+struct PropertyMetaInfo<
+    sycl::ext::intel::experimental::grf_size_automatic_key::value_t> {
+  static constexpr const char *name = "sycl-grf-size";
+  static constexpr unsigned int value = 0;
+};
+
+template <typename Properties>
+struct ConflictingProperties<sycl::ext::intel::experimental::grf_size_key,
+                             Properties>
+    : ContainsProperty<sycl::ext::intel::experimental::grf_size_automatic_key,
+                       Properties> {};
+
+template <typename Properties>
+struct ConflictingProperties<
+    sycl::ext::intel::experimental::grf_size_automatic_key, Properties>
+    : ContainsProperty<sycl::ext::intel::experimental::grf_size_key,
+                       Properties> {};
+
```

Comment:
yes, but do you mind if i do that in a separate pr? i made an internal issue tracker and added you to the notify list.

Reply:
No objections to that. My main point with the comment is to try and move all diagnostic messages to a compilation stage, which happens as early as possible, because it usually allows to emit better diagnostics. That `sycl-post-link` message you have doesn't even contain a kernel name for user to find the error.